File: IQ/Classes/Utility/LoggingList.vb

Public Class LoggingList(Of T)
    'Inherits List(Of String)
    Private Class clsData
        Public Message As String
        Public HeapSize As Double
        Public Time As Double
    End Class
    Private data As List(Of clsData) = New List(Of clsData)()
    Private sw As Stopwatch = New Stopwatch()
    Private StartBytes As Double

    Public Sub Clear()
        sw.Reset()
        data.Clear()
    End Sub
    Public Sub Start()
        sw.Start()
        StartBytes = System.GC.GetTotalMemory(True)
    End Sub
    Public Sub StopClock()
        sw.stop()
    End Sub
    Public Overloads Function ToList()
        Return data.ToList()
    End Function
    Public Overloads Sub Add(o As String)
        data.Add(New clsData() With {.HeapSize = Math.Round((System.GC.GetTotalMemory(False) - StartBytes) / (1024 ^ 2), 2), .Time = sw.ElapsedMilliseconds / 1000, .Message = o})
    End Sub
End Class

File: IQ/Classes/Utility/LoggingList.cs
public class LoggingList<T>
{
    public LoggingList()
    {
        // VBConversions Note: Non-static class variable initialization is below.  Class variables cannot be initially assigned non-static values in C#.
        data = new List<clsData>();
        sw = new Stopwatch();

    }
    //Inherits List(Of String)
    private class clsData
    {
        public string Message;
        public double HeapSize;
        public double Time;
    }
    private List<clsData> data; // VBConversions Note: Initial value cannot be assigned here since it is non-static.  Assignment has been moved to the class constructors.
    private Stopwatch sw; // VBConversions Note: Initial value cannot be assigned here since it is non-static.  Assignment has been moved to the class constructors.
    private double StartBytes;

    public void Clear()
    {
        sw.Reset();
        data.Clear();
    }
    public void Start()
    {
        sw.Start();
        StartBytes = System.Convert.ToDouble(System.GC.GetTotalMemory(true));
    }
    public void StopClock()
    {
        sw.stop();
    }
    public dynamic ToList()
    {
        return data.ToList();
    }
    public void Add(string o)
    {
        data.Add(new clsData() { HeapSize = Math.Round((System.GC.GetTotalMemory(false) - StartBytes) / (Math.Pow(1024, 2)), 2), Time = sw.ElapsedMilliseconds / 1000, Message = o });
    }
}